theseus/extlib/test_cusolver_lu_solver.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import pytest  # noqa: F401
+import torch  # needed for import of Torch C++ extensions to work
+from scipy.sparse import csr_matrix
+
+from theseus.extlib.sparse_matrix_utils import generate_sparse_matrix
```

Comment:
We should add most of these in the root init file so they are available with `import theseus as th`. Maybe you already do this in the next two PRs. We follow this import convention in examples/tutorials as well as tests.

Reply:
Sorry I think I don't understand this comment! So, the point of adding things in the __init__ is just to avoid one more step right? so we can call `th.generate_sparse_matrix`. However most of these functions (test utils and C++ extensions) look quite internal and meant to be exported to me, right? The solver class is very low level, essentially the smallest thing that can be exported to Python.

Reply:
Yes, so we can call `th.something` and the user doesn't have to remember which function is available in which submodule scope.

Ultimately, it depends if any of these functions would be useful to a user directly? If not we can leave that subset as is. Also if any of them are in fact 'internal' then maybe we should append a `_` in front of them.

Any thoughts, @luisenp?

Reply:
so, I would consider those functions to be hardly useful for the user. Of course the user may happen to be doing some low-level manipulation with batches sparse matrices and find those functions incidentally useful, but I would see adding those to the exported functionality more as a distraction than as something useful. Just my opinion!

Reply:
The matrix generation routine seems OK to have at `th` level, but for the rest I don't think so. From what I can tell, we still need to wrap the CUDA solver into a subclass of `th.optimizer.linear.LinearSolver`, and that's the one that should be exponsed at the `th` level. @maurimo can you confirm?

Reply:
oh sure the subclass of LinearSolver (PR24) is perfectly fine to be exported, I was referring to what is in extlib, which is too low-level to be exported directly.

Reply:
Sounds good, then let's leave as is. If we see a need later pull any of them up we can revisit.